### DISCUSSION TITLE
What about params Dictionary<string, object> initiative?

### DISCUSSION THREAD

**Discussion by iamkisly** [dotnet/csharplang#7970] [Q&A]
Hi guys. I found this earlier here. I haven't found a relevant ticket or discussion, but I would like to know if there is any news about the `params Dictionary<string, object>` initiative? This would be more convenient than `params KVPair<string, object>[]` for both deconstruction and initialization.

```cs
void Foo(int arg1, string arg2 = "boo", params Dictionary<string, string> kwargs){
    // do something 
}
Foo(12, hello="world", wednesday="mydude") 
```
-> 
`arg1=12; arg2="boo"; kwargs={["hello"]="world", ["wednesday"]="mydude"}`

But I found only one ticket in another repository [roslyn#4100](https://github.com/dotnet/roslyn/issues/4100).

**Comment by KennethHoff** [comment]
https://github.com/dotnet/csharplang/issues/7700 + https://github.com/dotnet/csharplang/issues/7822 = param dictionary?
Probably won't "fall out" of this, but maybe it starts the conversation.

**Comment by TahirAhmadov** [comment]
Once #7822 is implemented, you don't really need the `params` keyword - you will just do `Foo(12, ["hello": "world", "wednesday": "mydude"])` - basically, you just wrap it in `[ ]`, which is only 2 extra characters.
